usbinfo/darwin.py

```diff
@@ -18,17 +18,65 @@
 
 import copy
 import platform
+import re
 import subprocess
 
 from .posix import get_mounts
 
 # platform.mac_ver()'s first tuple element encodes the OS X version,
 # such as 10.11.3.
 OSX_VERSION_STR = platform.mac_ver()[0]
-OSX_VERSION_MAJOR_INT, \
-OSX_VERSION_MINOR_INT, \
-OSX_VERSION_MICRO_INT = [int(part) for part in OSX_VERSION_STR.split('.')]
-
+OSX_VERSION_INFO = [int(part) for part in OSX_VERSION_STR.split('.')]
+OSX_VERSION_MAJOR_INT = OSX_VERSION_INFO[0]
+OSX_VERSION_MINOR_INT = OSX_VERSION_INFO[1]
+OSX_VERSION_MICRO_INT = 0
+
+if len(OSX_VERSION_INFO) >= 3:
+    OSX_VERSION_MICRO_INT = OSX_VERSION_INFO[2]
+
+
+sanitize_pattern = re.compile(r'^(\s*?\<string\>)(.*?)(\<\/string\>.*?)$')
+
+def _sanitize_xml(data):
+    """Takes an plist (xml) and checks <string> defintions for control characters.
+
+       If a control character is found, the string is converted to hexidecimal.
+       For ST-Link devices, this properly displays the serial number, which 
+       is eroneously encoded as binary data, which causes the plistlib XML parser
+       to crash.
+
+       Returns the same document, with any mis-encoded <string> values converted 
+       to ascii hex."""
+    output = []
+
+    data = data.decode('utf-8')
```

Comment:
You are mutating data at the caller here. You could just encode it ad the for loop itself.